web/react/utils/text_formatting.jsx

```diff
@@ -3,21 +3,38 @@
 
 const Autolinker = require('autolinker');
 const Constants = require('./constants.jsx');
+const Markdown = require('./markdown.jsx');
 const UserStore = require('../stores/user_store.jsx');
 const Utils = require('./utils.jsx');
 
+const marked = require('marked');
+
+const markdownRenderer = new Markdown.MattermostMarkdownRenderer();
+
 // Performs formatting of user posts including highlighting mentions and search terms and converting urls, hashtags, and
 // @mentions to links by taking a user's message and returning a string of formatted html. Also takes a number of options
 // as part of the second parameter:
 // - searchTerm - If specified, this word is highlighted in the resulting html. Defaults to nothing.
 // - mentionHighlight - Specifies whether or not to highlight mentions of the current user. Defaults to true.
 // - singleline - Specifies whether or not to remove newlines. Defaults to false.
+// - markdown - Enables markdown parsing. Defaults to true.
 export function formatText(text, options = {}) {
-    let output = sanitizeHtml(text);
+    if (!('markdown' in options)) {
+        options.markdown = true;
+    }
+
+    // wait until marked can sanitize the html so that we don't break markdown block quotes
+    let output;
+    if (!options.markdown) {
+        output = sanitizeHtml(text);
+    } else {
+        output = text;
+    }
+
     const tokens = new Map();
 
     // replace important words and phrases with tokens
-    output = autolinkUrls(output, tokens);
+    output = autolinkUrls(output, tokens, !!options.markdown);
```

Comment:
It coalesces truthy and falsey values into actual true and false. I only used it here because it felt weird to possibly be passing undefined into a function.